exporter/otlpexporter/internal/arrow/exporter_test.go

```diff
@@ -277,30 +285,52 @@ func TestArrowExporterStreamFailure(t *testing.T) {
 // exercise the removeReady() code path.
 func TestArrowExporterStreamRace(t *testing.T) {
 	// Two streams ensures every possibility.
-	tc := newExporterTestCase(t, Noisy, twoStreamsSettings)
+	tc := newExporterTestCase(t, Noisy, Settings{
+		Enabled: true,
+
+		// This creates the conditions likely to produce a
+		// stream race in prioritizer.go.
+		NumStreams: 20,
+	})
+
+	var tries atomic.Int32
 
 	tc.streamCall.AnyTimes().DoAndReturn(tc.repeatedNewStream(func() testChannel {
 		tc := newUnresponsiveTestChannel()
 		// Immediately unblock to return the EOF to the stream
 		// receiver and shut down the stream.
 		go tc.unblock()
+		tries.Add(1)
 		return tc
 	}))
 
+	var wg sync.WaitGroup
+
 	bg := context.Background()
 	require.NoError(t, tc.exporter.Start(bg))
 
-	for tries := 0; tries < 1000; tries++ {
-		stream, err := tc.exporter.GetStream(bg)
-		require.NotNil(t, stream)
-		require.NoError(t, err)
-
-		err = stream.SendAndWait(bg, twoTraces)
+	callctx, cancel := context.WithCancel(bg)
 
-		require.Error(t, err)
-		require.True(t, errors.Is(err, ErrStreamRestarting))
+	// One goroutine will cancel the context in a while
+	// Goroutines will wait for the timeout.
```

Comment:
what's the default timeout period we're waiting for?

Reply:
Oh that comment was out of date with this change. Since there's no direct `getStream()` in the exporter, this had to be restructured. New comment is:

```
	// These goroutines will repeatedly try for an available
	// stream, but none will become available.  Eventually the
	// context will be canceled and cause these goroutines to
	// return.
```